core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss

```diff
@@ -0,0 +1,9 @@
+.site-name {@include icon('bars', $plugin_icon_colour, $margin: 1.2em)}
+
+.refinery_images_icon {@include icon('photo', $plugin_icon_colour, $margin: 1.2em)}
```

Comment:
There has to be some css _somewhere_ to link the class to the icon definition. We can change the class easily enough (from the first line below to the second), but it doesn't help us to get the icon on the page.  

```html
<a href="/refinery/pages" class="plugin bump-down refinery_pages_icon">Pages</a>
<a href="/refinery/pages" class="plugin bump-down files_icon">Pages</a>
```
You are right, it should be in the plugin, but be in `app/assets/plugin_name.css`, and add a `config/initializers/assets.rb` to add plugin_name.css to the asset pipeline. For example:

```scss
// refinerycms/pages/app/assets/refinery_pages.scss
.refinery_pages_icon {@include icon('files', $plugin_icon_colour, $margin: 1.2em)
```
----
An alternative is to create a class for each icon in the set, setting up a direct class<-->icon link.
The current class names are semantic classes, and the icon names are literal descriptions of the image.

```scss
.preview_icon {@include icon('eye', $icon_preview_colour)}
```
We could add 
```scss
.eye_icon {@include icon('eye')}
```
At that point we could use your original solution and write

```erb
  <%= link_to menu_item.title, refinery.url_for(menu_item.url), 
    class: ["plugin", "bump-down", "#{menu_item.icon}_icon"] << ("active" if menu_item.highlighted?(params))
  %>
```
to get

```html
<a href="/refinery/pages" class="plugin bump-down files_icon">Pages</a>
```